lib/src/script.dart

```diff
@@ -379,17 +379,18 @@ class _ScriptSpan extends SourceSpanMixin implements FileSpan {
   }
 
   SourceSpan union(SourceSpan other) {
-    if (other is! _ScriptSpan) return super.union(other);
+    if (other is _ScriptSpan) {
+      _ScriptSpan span = expand(other);
```

Comment:
In this case I'd use `as`, since it more clearly signals that `expand()` returns a broader type that we happen to know is a `_ScriptSpan`.